storybookjs/storybook
# Doc Blocks: Fix styling and parameter bugs

Closes #20850 



## What I did

This PR fixes a few outstanding bugs with docs and blocks.

1. Canvases in autodocs didn't respect `parameters.docs.canvas.withToolbar = false` - they do now.
2. We introduced a regression that broke Canvas when used in unattached mode and without any story children (deprecated API). This has now been fixed and has an internal story. This was reported here https://github.com/storybookjs/storybook/issues/19964#issuecomment-1405161473
4. inheritable styles like fonts were set on the docs content root, which would bleed through Unstyled blocks. We no longer set that reset, and it was a mistake it got set in the first place. This ensures that Unstyled removes styles from fonts.
5. Headers were not unstyled when wrapped in Unstyled. They are now. This happened because we still replace headers in MDX with custom components, and they had local styling. Now that local styling is removed so they rely solely on the global styles, which Unstyled can disable.

There's still a difference in header styles between autodocs and manual docs, specifically I've noticed that `h2` in autodocs have font-weight 400, while they have font-weight 700 in manual docs. font-weight 400 is a global style we set at the manager level via Emotion, and I can't figure out why that global style is not set in manual docs. It's not because it's overwritten, it is never there.
If anyone has inputs to this let me know.




## How to test

- See new internal story for Canvas in UI SB
- See the Unstyled docs page in UI SB
- See new stories for Stories and Primary blocks in UI SB



## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
